frontend/src/hooks/useDragItem.ts

```diff
@@ -3,7 +3,9 @@ import { useRef, useEffect, type RefObject } from 'react';
 export type OnMoveItem = (
     dragIndex: number,
     dropIndex: number,
-    save?: boolean,
+    save: boolean,
+    event: DragEvent,
+    draggedElement: HTMLElement,
```

Comment:
The extra parameters here is what allows us to do the same rapid oscillation prevention as we do for strategies.